soc/samr21: Initial support (#13494)
zephyrproject-rtos/zephyr: boards/arm/atsamr21_xpro/doc/index.rst

.. _atsamr21_xpro:

SAM R21 Xplained Pro Evaluation Kit
###################################

Overview
********

> The `SAMR21 Xplained Pro` is a compact evaluation board by Atmel featuring a

Reviewer: ```suggestion
The SAM R21 Xplained Pro is a compact evaluation board by Atmel featuring a
```

- One yellow user LED
- One mechanical user push button
- One reset button
- On-board USB based EDBG unit with serial console

Supported Features
==================

The atsamr21_xpro board configuration supports the following hardware
features:

+-----------+------------+--------------------------------------+
| Interface | Controller | Driver/Component                     |
+===========+============+======================================+
| NVIC      | on-chip    | nested vector interrupt controller   |
+-----------+------------+--------------------------------------+
| SYSTICK   | on-chip    | systick                              |
+-----------+------------+--------------------------------------+
| WDT       | on-chip    | Watchdog                             |
+-----------+------------+--------------------------------------+
| GPIO      | on-chip    | I/O ports                            |
+-----------+------------+--------------------------------------+
| USART     | on-chip    | Serial ports                         |
+-----------+------------+--------------------------------------+
| SPI       | on-chip    | Serial Peripheral Interface ports    |

> +-----------+------------+--------------------------------------+

Reviewer 2: Commit message states that "only UART, debug LED and user button are supported". Here we have also WDT, SPI.

Author: s/supported/tested/

     :align: center
     :alt: ATSAMR21-XPRO-pinout

Default Zephyr Peripheral Mapping:
----------------------------------
- SERCOM0 USART TX : PA5
- SERCOM0 USART RX : PA4
- SERCOM5 USART TX : PA23
- SERCOM5 USART RX : PA22
- SERCOM5 SPI MISO : PB16
- SERCOM5 SPI MOSI : PB22
- SERCOM5 SPI SCK  : PB23
- GPIO SPI CS      : PB17
- GPIO LED0        : PB17

System Clock
============

The SAMR21 MCU is configured to use the 32.768 kHz external oscillator
with the on-chip PLL generating the 48 MHz system clock.

Serial Port
===========

The SAMR21 MCU has six SERCOM based USARTs with two configured as USARTs in
this BSP. SERCOM0 is the default Zephyr console.

- SERCOM0 115200 8n1 connected to the onboard Atmel Embedded Debugger (EDBG)
- SERCOM5 115200 8n1

SPI Port
========

The SAMR21 MCU has 6 SERCOM based SPIs.

I2C Port
========

When connecting an I2C device and a logic analyzer to an I2C port at the same
time, the internal pull-up resistors are not sufficient for stable bus
operation. You probably have to connect external pull-ups to both bus lines. 10K
is a good value to start with.

Radio
=====

The SAMR21 SoC includes an on-chip AT86RF233 radio. It is internally

> connected via SPI and some GPIO pins and behaves in exact the same way as

Reviewer: ```suggestion
connected via SPI and some GPIO pins and behaves the same way as
```

+-------------+------------------------------------------------------------------------------------------+
| CS          | PB31 (OUT, GPIO output)                                                                  |
+-------------+------------------------------------------------------------------------------------------+
| IRQ         | PB00 (IN, GPIO external interrupt)                                                       |
+-------------+------------------------------------------------------------------------------------------+
| RSTN        | PB15 (OUT, GPIO output)                                                                  |
+-------------+------------------------------------------------------------------------------------------+
| SLP_TR      | PA20 (OUT, GPIO output)                                                                  |
+-------------+------------------------------------------------------------------------------------------+

Programming and Debugging
*************************

The SAM R21 Xplained Pro comes with a Atmel Embedded Debugger (EDBG).  This
provides a debug interface to the SAMR21 chip and is supported by
OpenOCD.

Flashing
========

> #. Build the Zephyr kernel and the ``hello_world`` sample application:

Reviewer: This change will link to the hello world sample doc:
```suggestion
#. Build the Zephyr kernel and the :ref:`hello_world` sample application:
```
